api/pkg/apis/projectcalico/v3/bgpconfig.go

```diff
@@ -80,6 +80,13 @@ type BGPConfigurationSpec struct {
 	// +kubebuilder:validation:Minimum:=1
 	// +kubebuilder:validation:Maximum:=65535
 	ListenPort uint16 `json:"listenPort,omitempty" validate:"omitempty,gt=0" confignamev1:"listen_port"`
+
+	// Optional BGP password for full node-to-mesh peerings.
```

Comment:
Lets call out in the comment something like so:

"This field can only be set on the default BGPConfiguration instance and requires that NodeMesh is enabled"..

Then in validation we can assert that is the case. Same for the restart time below.

Reply:
I think these should also both have `+optional` tags